File: Backend/AIBackend/Dominio/Model/PesquisaItemModel.cs
namespace AIBackend.Dominio.Model
{
    public class PesquisaItemModel
    {
        public int Id { get; set; }
        public int IdPesquisa { get; set; }
        public string Autores { get; set; }
        public bool IsPdfAvailable { get; set; }
        public string Sinopse { get; set; } = string.Empty;
        public string Titulo { get; set; } = string.Empty;
        public string UrlThumbnail { get; set; }

        public static explicit operator PesquisaItemModel(LivroViewModel viewModel)
        {
            return new PesquisaItemModel()
            {
                Autores = string.Join(",", viewModel.Autores),
                IsPdfAvailable = viewModel.IsPdfAvailable,
                Sinopse = viewModel.Sinopse,
                Titulo = viewModel.Titulo,
                UrlThumbnail = viewModel.UrlThumbnail
            };
        }
    }
}

File: ai_ui/lib/models/pesquisaItemModel.dart
class PesquisaItem {
  PesquisaItem({
    this.autores,
    this.isPdfAvailable,
    this.sinopse,
    this.titulo,
    this.urlThumbnail
  });

  String autores;
  bool isPdfAvailable;
  String sinopse;
  String titulo;
  String urlThumbnail;

  factory PesquisaItem.fromJson(Map<dynamic, dynamic> json){
    return PesquisaItem(
        autores: json["autores"] as String,
        isPdfAvailable: json["isPdfAvailable"] as bool,
        sinopse: json["sinopse"] as String,
        titulo: json["titulo"] as String,
        urlThumbnail: json["urlThumbnail"] as String);
  }
}